juju/testing/conn.go

```diff
@@ -253,6 +253,28 @@ func (s *JujuConnSuite) WaitForModelWatchersIdle(c *gc.C, modelUUID string) {
 	}
 }
 
+// EnsureCachedModel is used to ensure that the model specified is
+// in the model cache. This is used when tests create models and then
+// want to do things with those models where the actions may touch
+// the model cache.
+func (s *JujuConnSuite) EnsureCachedModel(c *gc.C, uuid string) {
+	start := time.Now()
```

Comment:
Is this clearer as a

```
stop := time.After(testing.LongWait)
retry := time.After(0)
for {
   select {
   case <-retry:
    _, err := s.Controller.Model(uuid)
    if err == nil {
     return
   }
   timeout = time.After(testing.ShortWait)
   case <-stop:
    c.Errorf("model %v not seen in cache after %v", uuid, testing.LongWait)
    return
 }
}
```

Reply:
Yes, I think something with a select is better.